crates/bevy_gizmos/src/grid.rs

```diff
@@ -368,78 +368,71 @@ fn draw_grid<Config, Clear>(
         return;
     }
 
+    #[inline]
+    fn or_zero(cond: bool, val: Vec3) -> Vec3 {
+        if cond {
+            val
+        } else {
+            Vec3::ZERO
+        }
+    }
+
     // Offset between two adjacent grid cells along the x/y-axis and accounting for skew.
-    let dx = spacing.x
-        * Vec3::new(1., skew.y.tan(), skew.z.tan())
-        * if cell_count.x != 0 { 1. } else { 0. };
-    let dy = spacing.y
-        * Vec3::new(skew.x.tan(), 1., skew.z.tan())
-        * if cell_count.y != 0 { 1. } else { 0. };
-    let dz = spacing.z
-        * Vec3::new(skew.x.tan(), skew.y.tan(), 1.)
-        * if cell_count.z != 0 { 1. } else { 0. };
+    let skew_tan = Vec3::from(skew.to_array().map(f32::tan));
```

Comment:
In light of recent changes, f32::tan should probably be changed to ops::tan, I believe.

Edit: nvm, this is gizmo code so it doesn't matter.